src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/TaskManagementService.java

```diff
@@ -372,11 +375,42 @@ public void cancelTask(String taskId,
         if (isRelease2EndpointsFeatureEnabled) {
             //Lock & update Task
             TaskResource task = findByIdAndObtainLock(taskId);
+            CFTTaskState previousTaskState = task.getState();
             task.setState(CFTTaskState.CANCELLED);
-            //Perform Camunda updates
-            camundaService.cancelTask(taskId);
-            //Commit transaction
-            cftTaskDatabaseService.saveTask(task);
+
+            boolean isCftTaskStateExist = camundaService.isCftTaskStateExistInCamunda(taskId);
+
+            log.info("{} previousTaskState : {} - isCftTaskStateExist : {}",
+                taskId, previousTaskState, isCftTaskStateExist);
+
+            try {
+                //Perform Camunda updates
+                camundaService.cancelTask(taskId);
+                log.info("{} cancelled in camunda", taskId);
+                //Commit transaction
+                cftTaskDatabaseService.saveTask(task);
+                log.info("{} cancelled in CFT", taskId);
+            } catch (TaskCancelException ex) {
+                log.info("{} an error occurred when cancelling task. Checking Cft Task State to sync DBs", taskId);
+                if (isCftTaskStateExist) {
+                    log.info("{} TaskCancelException occurred due to cftTaskState exists in Camunda history", taskId);
+                    throw new TaskCancelException(TASK_CANCEL_UNABLE_TO_CANCEL);
+                }
+
+                if (asList(CFTTaskState.ASSIGNED, CFTTaskState.UNASSIGNED,
+                    CFTTaskState.COMPLETED, CFTTaskState.CANCELLED).contains(previousTaskState)) {
+                    task.setState(CFTTaskState.TERMINATED);
+                    cftTaskDatabaseService.saveTask(task);
+                    log.info("{} cftTaskState updated TERMINATED", taskId);
```

Comment:
👍

Reply:
This is probably simpler if we do if CFTTaskState is not Terminated then update.